### ISSUE TITLE
Can't move on stairs / slaps

### ISSUE THREAD

**Issue by SURVAml** [DragonetMC/DragonProxy#53]
First: It's a really cool project, it's unbelievable :)
There's a problem when walking on stairs / slaps, you can't move because it always teleports you back.

**Comment by MrPowerGamerBR** [comment]
Or it is a issue with the Server AntiCheat (I don't think it is) or it is because the slab height is a bit different than the PC slabs.

If it is the second issue, then DragonProxy needs to get the move packet, verify if the block below the client is a slab, and, if it is, increase/decrease the height of the player when sending the packet.

**Comment by RandomAltThingy** [comment]
I think it's because the client does not see it as a non cube block, so it
says that it is inside the block and glitches you out. This is just what
**i** think

**Comment by MrPowerGamerBR** [comment]
There is a way to fix it if you have the block below the player when translating the packets.

The problem is that DragonProxy doesn't save the player chunks anywhere.

**Comment by DefinitlyEvil** [comment]
@SURVAml You should jump onto it rather than walking straight onto it.

**Comment by MrPowerGamerBR** [comment]
@DefinitlyEvil yeah, but we should fix the problem instead of trying workarounds, right?

This happens with carpets too, this happens because of the height level of the blocks, carpets in 1.8 are a bit taller, but in MCPE you don't stay higher when stepping on a carpet, this causes you to go back.

**Comment by DefinitlyEvil** [comment]
Idk how to solve this unless we cache the map on the server but that will eat RAM.

**Comment by MrPowerGamerBR** [comment]
You don't need to cache the entire map, you just need to cache the chunk
where the player is standing.
Em 17 de jan de 2016 11:01 PM, "The Evil Man" <notifications@github.com>
escreveu:

**Comment by DefinitlyEvil** [comment]
But server won't send chunks again which already sent.

**Comment by MrPowerGamerBR** [comment]
So... Yeah, the only solution is caching the loaded chunks.

You don't need to cache for every single player tho, just create a global
cache map.

Another way is saving the coords where slabs/carpets/stairs are when the
chunk is sent, then we just compare the chunks on the move packet.
Em 17 de jan de 2016 11:04 PM, "The Evil Man" <notifications@github.com>
escreveu:

**Comment by DefinitlyEvil** [comment]
@MrPowerGamerBR Seems good but that will eat both RAM and CPU.
